connectors-native-sdk-java/README.md

```diff
@@ -0,0 +1,131 @@
+  * core.sql -  this file creates all required base roles, schemas, and database objects (base features) that are required by particular optional features.
+  * application/ - this directory includes files that create database objects for functionalities like configuration, ingestion, observability, lifecycle, and scheduler.
+  * task_reactor.sql - the file creates all database objects that are the API for task reactor instances management.
+
+### Tests
+
+Every quality project should have tests, and Native SDK for Connectors Java is no different. The SDK is tested by tests 
+at different layers, which are:
+* Unit tests - lightweight tests that test the logic of the Java code in the separation from the Snowflake environment. They can be found in the `src/tests` directory.
+* Integration tests - these tests most often test code that is responsible for operations in the Snowflake environment, so they need a Snowflake account connection in order to be executed. These tests create a separate database that is used during tests run. They can be found in the `src/intTests` directory.
+* Application tests - the main difference between these tests and integration tests is that these tests test the whole flow of the particular functionality in the environment of the Snowflake Native Application instead of the database. It's required to use Native Application as a test environment because sometimes, Native Applications behavior differs from the standard Snowflake database. These tests can be found in the `src/appTests` directory.
+
+As for integration and application tests they require connection to Snowflake. Connection details can be defined using 
+the `.env/snowflake_credentials file`. Application tests directory also contains an empty connector application in 
+`src/appTests/resources`. That application is deployed during the test suite execution.
+
+The code is tested with the usage of the Native SDK for Connectors Java Test library which consists of Java utils 
+helping in testing the developed Connectors. These utils are custom assertions and in-memory objects (mocks) for Java 
+classes used in the Native Java SDK.
+
+## Application areas supported by the SDK
+
+The Native SDK for Connectors helps developers cover many crucial connector application areas by delivering appropriate 
+mechanisms (tables, views, procedures etc.). Among these areas we can mention:
+* Configuration (Wizard) - the SDK defines the division of connector configuration flow into 3 main steps where logically different configurations are provided:
+  * Prerequisites - The prerequisites step is the first step of the wizard phase of the connector. This step is completely optional, but it is recommended, if the end user needs to perform some additional setup outside of the native app or even the whole Snowflake context. An example of this could be setting up authentication and authorization in the source system from which the data will be ingested.
```

Comment:
```suggestion
  * Prerequisites - The prerequisites step is the first step of the wizard phase of the connector. This step is completely optional, but it is recommended, if the end user needs to perform some additional setup outside of the Native App or even the whole Snowflake context. An example of this could be setting up authentication and authorization in the source system from which the data will be ingested.
```